lib/ofp-util.c

```diff
@@ -1884,6 +1884,109 @@ ofputil_encode_update_map(enum ofp_version ofp_version,
     return request;
 }
 
+enum ofperr
+ofputil_decode_dump_map_request(struct ol_dump_map_request *msg,
+                                const ovs_be16 **map_ids,
+                                const struct ofp_header *oh)
+{
+    enum ofperr error = 0;
+    struct ofpbuf b = ofpbuf_const_initializer(oh, ntohs(oh->length));
+    enum ofpraw raw = ofpraw_pull_assert(&b);
+    if (raw != OFPRAW_NXT_DUMP_MAP_REQUEST) {
+        return OFPERR_OFPBMC_BAD_TYPE;
+    }
+
+    struct ol_dump_map_request *buffer = ofpbuf_pull(&b,
+            sizeof(struct ol_dump_map_request));
+
+    msg->filter_prog = ntohs(buffer->filter_prog);
+    msg->nb_maps = ntohs(buffer->nb_maps);
+
+    if(!msg->filter_prog) {
+        VLOG_WARN_RL(&bad_ofmsg_rl,
+                     "The filter program identifier is not provided.");
+        return OFPERR_OFPBRC_EPERM;
+    }
+
+    size_t map_ids_size = (size_t)(sizeof(**map_ids) * msg->nb_maps);
+    *map_ids = ofpbuf_try_pull(&b, map_ids_size);
+    if (!*map_ids) {
+        VLOG_WARN_RL(&bad_ofmsg_rl, "size of provided map identifiers array is incorrect"
+                                    " (%"PRIu32")", map_ids_size);
+        return OFPERR_OFPBMC_BAD_LEN;
+    }
+
+    return error;
+}
+
+struct ofpbuf *
+ofputil_encode_dump_map_request(enum ofp_version ofp_version,
+                                const ovs_be16 prog,
+                                const ovs_be16 nb_maps,
+                                const ovs_be16 *map_ids)
+{
+    struct ofpbuf *request;
+    struct ol_dump_map_request *msg;
+    size_t map_ids_size = (size_t)(sizeof(*map_ids) * nb_maps);
+
+    request = ofpraw_alloc(OFPRAW_NXT_DUMP_MAP_REQUEST, ofp_version, map_ids_size);
+    ofpbuf_put_zeros(request, sizeof *msg);
+
+    msg = request->msg;
+    msg->filter_prog = htons(prog);
+    msg->nb_maps = htons(nb_maps);
+
+    ofpbuf_put(request, map_ids, map_ids_size);
+
+    ofpmsg_update_length(request);
+
+    return request;
+}
+
+struct ofpbuf *
+ofputil_encode_dump_map_reply(struct ol_dump_map_request *msg,
+                              const struct ofp_header *oh,
+                              const struct ubpf_map **map,
+                              const ovs_be16 *map_ids,
+                              void **data,
+                              unsigned int *nb_elems)
+{
+    struct ofpbuf *output_buffer;
+    struct ol_dump_map_reply *dump_map_reply;
+
+    size_t all_maps_size = 0;
+    for(int i = 0; i < msg->nb_maps; i++) {
+        all_maps_size += (size_t) sizeof(struct ol_dump_map);
```

Comment:
You don't need the cast before the `sizeof`, no?

Reply:
Yes. Done.